### PR TITLE
Subscription authorization

### PR DESCRIPTION
**Pull request by rcillo**
https://techjira.zalando.net/browse/ARUHA-1681

Since the introduction of Event Type Authorization, it was planned that
Subscriptions would also be authorized. Even though data is already
protected through Event Type Authorization, users can still accidentally
commit offsets to subscriptions owned by other users, eventually causing
applications to not process all the data. Subscriptions could also be
accidentally deleted or have its slots assigned to some other
application impeding applications from consuming data.

With this feature we address all of the problems above. This is just the
API definition.

### CODE REVIEW

File: docs/_data/nakadi-event-bus-api.yaml [hunk 1]
```diff
@@ -1109,6 +1109,10 @@ paths:
                   $ref: '#/definitions/CursorCommitResult'
             required:
               - items
+        '403':
```
Comment (v-stepanov) [inline docs/_data/nakadi-event-bus-api.yaml]: can you please also add it to reset cursors endpoin

Comment (v-stepanov) [reply docs/_data/nakadi-event-bus-api.yaml]: and also to delete endpoint

**Comment by v-stepanov** [comment]
I'm missing update-subscription endpoint :)

**Comment by rcillo** [comment]
@v-stepanov addressed the comments.

File: docs/_data/nakadi-event-bus-api.yaml [hunk 2]
```diff
@@ -1019,6 +1019,29 @@ paths:
...
+        '404':
+          description: Subscription not found
+          schema:
+            $ref: '#/definitions/Problem'
+    put:
+      tags:
+        - subscription-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: Updates a subscription.
```
Comment (v-stepanov) [inline docs/_data/nakadi-event-bus-api.yaml]: Do you think it makes sense to describe here which parameters of subscription it is possible to update? I'm afraid users will try to update owning app and list of event-types.

File: docs/_data/nakadi-event-bus-api.yaml [hunk 3]
```diff
@@ -1019,6 +1019,29 @@ paths:
...
+      parameters:
```
Comment (v-stepanov) [inline docs/_data/nakadi-event-bus-api.yaml]: I think you also need to describe here the subscription itself that comes in body

File: docs/_data/nakadi-event-bus-api.yaml [hunk 4]
```diff
@@ -1019,6 +1019,38 @@ paths:
...
+        This endpoint only allows to update the authorization section of a subscription. All other properties are
+        immutable. This operation is restricted to subjects with administrative role.
+      parameters:
+        - name: subscription
+          in: body
+          description: Subscription to create
+          schema:
+            $ref: '#/definitions/Subscription'
+          required: true
+      parameters:
```
Comment (v-stepanov) [inline docs/_data/nakadi-event-bus-api.yaml]: you have `parameters` properties doubled

File: docs/_data/nakadi-event-bus-api.yaml [hunk 5]
```diff
@@ -1019,6 +1019,38 @@ paths:
...
+        - subscription-api
+      security:
+        - oauth2: ['nakadi.event_stream.read']
+      description: |
+        This endpoint only allows to update the authorization section of a subscription. All other properties are
+        immutable. This operation is restricted to subjects with administrative role.
+      parameters:
+        - name: subscription
+          in: body
+          description: Subscription to create
```
Comment (v-stepanov) [inline docs/_data/nakadi-event-bus-api.yaml]: not to create

**Comment by v-stepanov** [comment]
@rcillo Now everything looks good for me. The only thing that can be improved - is to specify which role is needed to reset cursors. I think now it will be not clear for the user reading the API description which role he needs to reset cursors.

File: docs/_data/nakadi-event-bus-api.yaml [hunk 6]
```diff
@@ -1019,6 +1019,41 @@ paths:
...
+          type: string
+          description: Id of subscription
+          required: true
+      responses:
+        '204':
+          description: Subscription was updated
+        '403':
+          description: Access forbidden
+          schema:
+            $ref: '#/definitions/Problem'
```
Comment (antban) [inline docs/_data/nakadi-event-bus-api.yaml]: But where is 
```
         '400':
           description: Bad Request
           schema:
             $ref: '#/definitions/Problem'
```

**Pull request merged**